### PR TITLE
Add --exclude-folders option to skip specified folders

### LINKED ISSUE #36: Option to ignore a list of folders?
Some IMAP servers provides non-e-mails as mailboxes. For instance Exchange has:

* _Calendar_
* _Contacts_
* _Tasks_
* _RSS Feeds_
* _Journal_
* Also: _Sync Issues_

When backuping such mailboxes with _imapbackup_ it can fails, or even if it does not, maybe you don't want to backup theses folders.

There is an option to pass folder(s) to backup, but none for the one to ignore

If desired I can provide a small PR introducing the `--ignored-folders` option with the following changes:

```Diff
@@ -675,12 +675,16 @@ def main():
          server = connect_and_login(config)
          names = get_names(server, config['compress'], config['thunderbird'],
                            config['nospinner'])
+         ignored_dirs = []
          if config.get('folders'):
              dirs = map(lambda x: x.strip(), config.get('folders').split(','))
              if config['thunderbird']:
                  dirs = [i.replace("Inbox", "INBOX", 1) if i.startswith("Inbox") else i
                          for i in dirs]
              names = filter(lambda x: x[0] in dirs, names)
+         elif config.get('ignored-folders'):
+             # Use "--ignored-folders" option's list only if "--folders" option was not set
+             ignored_dirs = map(lambda x: x.strip(), config.get('ignored-folders').split(','))
  
          # for n, name in enumerate(names): # *DEBUG
          #   print n, name # *DEBUG
@@ -690,6 +694,11 @@ def main():
          for name_pair in names:
              try:
                  foldername, filename = name_pair
+ 
+                 # Skip ignored directory
+                 if foldername in ignored_dirs:
+                     continue
+ 
                  fol_messages = scan_folder(
                      server, foldername, config['nospinner'])
                  fil_messages = scan_file(filename, config['compress'],
```

### PR DESCRIPTION
**Pull request by samsonjs**
Based on code in #36 by @C-Duv. Closes #36

### CODE REVIEW

File: imapbackup38.py [hunk 1]
```diff
@@ -398,9 +398,10 @@ def print_usage():
     print (" -d DIR --mbox-dir=DIR         Write mbox files to directory. (defaults to cwd)")
     print (" -a --append-to-mboxes         Append new messages to mbox files. (default)")
     print (" -y --yes-overwrite-mboxes     Overwite existing mbox files instead of appending.")
-    print (" -f FOLDERS --folders=FOLDERS  Specifify which folders use.  Comma separated list.")
+    print (" -f FOLDERS --folders=FOLDERS  Specify which folders to include. Overrides --exclude-folders. Comma separated list.")
+    print (" --exclude-folders=FOLDERS     Specify which folders to exclude. Overridden by --folders. Comma separated list.")
```
Comment (dumblob) [inline imapbackup38.py:402]: How does this PR support folders with commans in their names (I've seen such)?

File: imapbackup38.py [hunk 2]
```diff
@@ -643,13 +646,17 @@ def main():
...
         if config.get('folders'):
+            # Takes precedence over --exclude-folders
             dirs = list(map(lambda x: x.strip(), config.get('folders').split(',')))
             if config['thunderbird']:
                 dirs = [i.replace("Inbox", "INBOX", 1) if i.startswith("Inbox") else i
                         for i in dirs]
             names = list(filter(lambda x: x[0] in dirs, names))
-
+        elif config.get('exclude-folders'):
+            # Use --exclude-folders option only when --folders is not set
```
Comment (dumblob) [inline imapbackup38.py:658]: This must not be silent - if both appear on the command line, it must fail immediately and not silently do any internal precedence handling.

**Comment by dumblob** [comment]
Except for the comments above, I think this does the job. Thanks!

**Comment by C-Duv** [comment]
Thanks samsonjs for the PR :)

**Comment by rcarmo** [comment]
This looks nice, if you fix the silent override thing I’ll merge it on my next inbox dive…

File: imapbackup38.py (hunk 1 above)
Comment (samsonjs) [reply imapbackup38.py:402]: It doesn't, it behaves just like the existing --folders option and they should match each other. I hadn't thought about that case though. Some languages let you use any character to delimiter things like a regex which is nice if you need to match something with a bunch of slashes for example. Having a configurable delimiter might be a good way to solve it, maybe something like `--folder-delimiter='$' --folders='Inbox$Sent$Archive'`?

I feel like changing how --folders works is probably a separate issue/PR though.

File: imapbackup38.py (hunk 2 above)
Comment (samsonjs) [reply imapbackup38.py:658]: Good call, thanks. I'll make it fail instead.

Comment (samsonjs) [reply imapbackup38.py]: Fixed by c7a53d6

File: imapbackup38.py (hunk 1 above)
Comment (dumblob) [reply imapbackup38.py]: > I feel like changing how --folders works is probably a separate issue/PR though.

:+1:

**Review by dumblob — APPROVED** [review]

**Pull request merged**